graylog2-server/src/main/java/org/graylog/security/UserAuthorizer.java

```diff
@@ -0,0 +1,56 @@
+/**
+ * This file is part of Graylog.
+ *
+ * Graylog is free software: you can redistribute it and/or modify
+ * it under the terms of the GNU General Public License as published by
+ * the Free Software Foundation, either version 3 of the License, or
+ * (at your option) any later version.
+ *
+ * Graylog is distributed in the hope that it will be useful,
+ * but WITHOUT ANY WARRANTY; without even the implied warranty of
+ * MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.  See the
+ * GNU General Public License for more details.
+ *
+ * You should have received a copy of the GNU General Public License
+ * along with Graylog.  If not, see <http://www.gnu.org/licenses/>.
+ */
+package org.graylog.security;
+
+import com.google.inject.assistedinject.Assisted;
+import org.apache.shiro.mgt.DefaultSecurityManager;
+import org.apache.shiro.subject.SimplePrincipalCollection;
+import org.apache.shiro.subject.Subject;
+import org.graylog.grn.GRN;
+
+import javax.inject.Inject;
+
+import static com.google.common.base.Preconditions.checkArgument;
+import static org.apache.commons.lang3.StringUtils.isNotBlank;
+
+public class UserAuthorizer {
+    public interface Factory {
+        UserAuthorizer create(GRN grantee);
+    }
+
+    private final Subject subject;
+
+    @Inject
+    public UserAuthorizer(DefaultSecurityManager securityManager, @Assisted GRN grantee) {
+        this.subject = new Subject.Builder(securityManager)
+                .principals(new SimplePrincipalCollection(grantee.entity(), "UserAuthorizer"))
```

Comment:
This will potentially build a "user" with the principal of a team..
We need to solve this sooner or later..

Reply:
I refactored the code to also handle teams in the future.